tests/recv.py

```diff
@@ -0,0 +1,219 @@
+import asyncio
+import os
+
+from distributed.comm.utils import from_frames
+from distributed.utils import nbytes
+
+import cloudpickle
+import numpy as np
+import pynvml
+import pytest
+import ucp
+
+# import subprocess
+
+
+pynvml.nvmlInit()
+
+
+cmd = "nvidia-smi nvlink --setcontrol 0bz"  # Get output in bytes
+# subprocess.check_call(cmd, shell=True)
+
+pynvml = pytest.importorskip("pynvml", reason="PYNVML not installed")
+ITERATIONS = 1000
+
+
+def cuda_array(size):
+    import cupy as cp
+
+    return cp.empty(size, dtype=cp.uint8)
+    # return rmm.device_array(size, dtype=np.uint8)
```

Comment:
@madsbk in distributed this function normally return rmm/numba.device_array .  When sending/receiving cupy array is this safe ?